### ISSUE TITLE
Incorrectly ignore image with .PNG extension ('NoneType' object has no attribute 'seek')

### ISSUE THREAD

**Issue by viktorpovazhuk** [ultralytics/ultralytics#4403]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

YOLOv8 can't read images with '.PNG' extension.

![image](https://github.com/ultralytics/ultralytics/assets/77798403/7ca076e6-2f56-488b-9dcc-e1d0f98424c2)

### Environment

- YOLO: YOLO 8.0.155
- OS: Windows 64

### Minimal Reproducible Example

```
from pathlib import Path
from ultralytics import YOLO

data_path = Path('~/data/sample/data.yaml')

model = YOLO('yolov8s.pt')

results = model.train(data=str(data_path), epochs=1, imgsz=640)
```

Images are attached.

[sample.zip](https://github.com/ultralytics/ultralytics/files/12359070/sample.zip)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by cparcerisas** [comment]
Hello, I found the same bug (same version, same OS). In version 8.0.150 it works fine. 
Exception is raised by the function exif_size on ultralytics.data.utils 

Thanks!

**Comment by RTaar** [comment]
Hello! Same bug on ubuntu 22.04

**Comment by glenn-jocher** [comment]
@viktorpovazhuk @cparcerisas @RTaar 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀.  PNG training works correctly as verified with the GlobalWheat2020 dataset, which is all PNG images:

```
yolo train model=yolov8n.pt data=GlobalWheat2020.yaml epochs=1
```

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by rmukh** [comment]
Same.

Works correctly on 8.0.154. Ubuntu 20 and 22.

**Comment by glenn-jocher** [comment]
@RTaar @rmukh @cparcerisas @viktorpovazhuk hi guys, thanks for the bug report, I'm able to reproduce, working on a fix now.

Seems like we need some *.png training in our tests also.

**Comment by glenn-jocher** [comment]
@RTaar @rmukh @cparcerisas @viktorpovazhuk great news 😃! Your original issue may now be resolved ✅ in PR #4406. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by jronmi** [comment]
I found this same issue with .tif / .tiff files as well

**Comment by RTaar** [comment]
> @RTaar @rmukh @cparcerisas @viktorpovazhuk great news 😃! Your original issue may now be resolved ✅ in PR #4406.
> 
> To get this update:
> 
> * **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
> * **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
> * **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  [![Run on Gradient](https://camo.githubusercontent.com/c9931a1689c37ab786edd3e1e5f59b9a6f7d097628c4689ce2432563ef884524/68747470733a2f2f6173736574732e706170657273706163652e696f2f696d672f6772616469656e742d62616467652e737667)](https://console.paperspace.com/github/ultralytics/ultralytics) [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/ultralytics/yolov8)
> * **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image [![Docker Pulls](https://camo.githubusercontent.com/7401823bed1473d6db1ff60b933d51cd1534c448488f06c446c322227d31d966/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f756c7472616c79746963733f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/ultralytics)
> 
> Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

Hi @glenn-jocher. 

Bug fixed. Ubuntu 22.04,  pip ''8.0.156'' 

Thanks!

**Comment by xumix** [comment]
Looks like the issue is related: https://github.com/ultralytics/ultralytics/issues/4496
I think more robust gif/png image format handling and logging would be great. 
Maybe an option to ignore or fail when unknown format is found would be even better

**Comment by glenn-jocher** [comment]
@xumix unknown formats are ignored. Supported formats are here: https://docs.ultralytics.com/modes/predict/#image-and-video-formats

**Comment by xumix** [comment]
> @xumix unknown formats are ignored. Supported formats are here: https://docs.ultralytics.com/modes/predict/#image-and-video-formats

issue #4496 actually proves you wrong, I was able to find the image which is a GIF saved as image.jpg, so an actual format check (not an extension check) with a descriptive message would be so helpful

**Comment by glenn-jocher** [comment]
Thanks for your feedback, @xumix. In our current implementation, the method to identify file formats indeed primarily focuses on the file extension. Your suggestion to examine the files more deeply than just by extension and provide more descriptive messaging when unsupported formats are encountered is helpful. We take all our user feedback seriously, we will consider this in our future updates.

Thank you for being an active member of our YOLOv8 community and helping to improve the user experience. Your patience and understanding are greatly appreciated!

**Comment by xumix** [comment]
Thanks you for your efforts and fast response!

**Comment by glenn-jocher** [comment]
@xumix you're welcome! I'm glad to hear that our team's assistance was helpful to you. We aim to ensure the best performance and usability for YOLOv8 and really appreciate user feedback for improvements. If you encounter any other issues, do not hesitate to let us know. We're here to help. Happy coding!
